provider/ansible.rb

```diff
@@ -313,6 +321,68 @@ def generate_typed_array(type, data) end
       def emit_nested_object(data) end
 
       def emit_resourceref_object(data) end
+
+      # rubocop:disable Metrics/AbcSize
```

Comment:
I'm pretty willing to agree with rubocop that this method could use being split up.

Reply:
Likewise. I haven't reviewed it yet because I'm having a bit of trouble parsing what's going on. Maybe making some extra functions/variables/comments will help clear it up :)

Reply:
I added some comments. I can rewrite if you'd like but trying to avoid.

Context: This function used to exist in api/type.rb. I feel it doesn't belong there since it's primarily used by Ansible + test_data generation. The function hasn't really changed since api/type.rb

Reply:
Sorry - these comments aren't really helping me yet.  What's the input and output of the method, for the first thing?  `resourcerefs_for_properties`... so it takes in a list of properties and returns a list of `ResourceRef` objects?  Is that `ResourceRef` or `IndividualResourceRef`?

Reply:
With the rename, hopefully that helps make this clearer.

I'm happy to refactor this if we want to. I didn't want to fiddle with this code too much in this PR because it's a move at this moment. There was code in api/resource.rb that was only being used in Ansible + test_data generation. Didn't seem appropriate to keep it there.

Reply:
Ah, that comment did help... I'm fine with this, then.